crates/common/trie/trie_sorted.rs

```diff
@@ -0,0 +1,451 @@
+use crate::{
+    EMPTY_TRIE_HASH, Nibbles, Node, NodeHash, TrieDB, TrieError,
+    node::{BranchNode, ExtensionNode, LeafNode},
+};
+use crossbeam::channel::{Receiver, Sender, bounded};
+use ethereum_types::H256;
+use ethrex_threadpool::ThreadPool;
+use std::{sync::Arc, thread::scope};
+use tracing::debug;
+
+#[derive(Debug, Default, Clone)]
+struct StackElement {
+    path: Nibbles,
+    element: BranchNode,
+}
+
+// The large size isn't a performance problem because we use a single instance of this
+// struct
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone)]
+enum CenterSideElement {
+    Branch { node: BranchNode },
+    Leaf { value: Vec<u8> },
+}
```

Comment:
Not to handle necessarily for this PR, but having `Leaf` contain a `Vec` might force unnecessary allocations, specially if we replace a `Leaf` with a `Branch` and then a `Leaf` again, we'll free the first vector and reallocate afterwards.